addons/nuxeo-segment.io-connector/src/main/java/org/nuxeo/segment/io/SegmentIOMapper.java

```diff
@@ -78,32 +81,41 @@ public boolean isScreen() {
         return ACTIONS.screen.name().equalsIgnoreCase(target);
     }
 
-    public Map<String, Serializable> getMappedData(Map<String, Object> context) {
+    public Class<?> getGroovyClazz() {
+        if (klass == null) {
+            synchronized (this) {
+                // Define Groovy class based on mapper contributions
```

Comment:
A second nullity check should be done to avoid re-creating the resource by the threads which were waiting on the synchronized block.